### DISCUSSION TITLE
dnscrypt-proxy issue on macos Monterey 12.3.1

### DISCUSSION THREAD

**Discussion by iam-01** [DNSCrypt/dnscrypt-proxy#2097] [Q&A]
Dear all,

since upgrade to macOS Monterey 12.3.1 running dnscrypt-proxy as a service does not working properly any longer. Repetitively the message "dnscrypt-proxy was blocked from use because it is not from an identified developer" occurs, even  if the allowance was granted several times,  please see attachment. This behavior also impacts the applications xbar & dnscrypt-proxy switch.
In manual mode via terminal everything works fine, although a permanent open root-shell is necessary, which from a security-perspective is extremely suspicious.

Therefor my question: Do you have any advice or hint to fix the behavior outlined above?

Many thanks in advance.

[dnscrypt-proxy_err_20220417_01.pdf](https://github.com/DNSCrypt/dnscrypt-proxy/files/8502086/dnscrypt-proxy_err_20220417_01.pdf)

**Comment by jedisct1** [comment]
What does the following command print?

```sh
shasum $(which dnscrypt-proxy)
```

**Reply by iam-01** [reply]
The output of this command is: 6b055fa0e93605e1e04b19b78bb94e6e1548b5e6  /opt/DNSCrypt/dnscrypt-proxy
By the way, I've downloaded and re-installed the latest version 2.1.1. Unfortunately without success.

**Comment by jedisct1** [comment]
That file is signed. You can check its signature with;

```sh
codesign -dv $(which dnscrypt-proxy)
```

Make sure that you don't have another version installed somewhere else, and also installed as a service, or started using a different procedure.

**Reply by iam-01** [reply]
Clean installation performed and checked already, no changes in described behavior.
In the meantime I experienced an alternative installation method via Macports and by deleting the direct / native installation. This works now for me. Thank's for your support.
